fixture-monkey-tests/java-17-tests/src/test/java/com/navercorp/fixturemonkey/tests/java17/JdkVariantOptionsTest.java

```diff
@@ -0,0 +1,37 @@
+package com.navercorp.fixturemonkey.tests.java17;
+
+import static com.navercorp.fixturemonkey.tests.TestEnvironment.TEST_COUNT;
+import static org.assertj.core.api.BDDAssertions.then;
+
+import java.util.List;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.RepeatedTest;
+
+import com.navercorp.fixturemonkey.FixtureMonkey;
+import com.navercorp.fixturemonkey.api.generator.InterfaceObjectPropertyGenerator;
+import com.navercorp.fixturemonkey.api.introspector.ConstructorPropertiesArbitraryIntrospector;
+import com.navercorp.fixturemonkey.tests.java17.SealedClassTestSpecs.BaseSealedClass;
+import com.navercorp.fixturemonkey.tests.java17.SealedClassTestSpecs.SealedClass;
+
+class JdkVariantOptionsTest {
+	private static final FixtureMonkey SUT = FixtureMonkey.builder()
+		.objectIntrospector(ConstructorPropertiesArbitraryIntrospector.INSTANCE)
+		.defaultNotNull(true)
+		.pushAssignableTypeObjectPropertyGenerator(
+			BaseSealedClass.class,
+			new InterfaceObjectPropertyGenerator<>(
+				List.of(SealedClass.class)
+			)
+		)
+		.build();
+
+	@RepeatedTest(TEST_COUNT)
+	@DisplayName("User-added ObjectPropertyGenerators have a higher priority.")
+	void sampleSealedClass() {
+		// when
+		BaseSealedClass actual = SUT.giveMeOne(BaseSealedClass.class);
```

Comment:
In this case, I think a `Fixture Monkey` instance should be in the test case.

It is a crucial information that Fixture Monkey use `pushAssignableTypeObjectPropertyGenerator` option.